docs/DeveloperGuide.md

```diff
@@ -1,38 +1,228 @@
+| v1.0    | forgetful user   | view my expenses                                    | remove incorrect or outdated entries from my transaction history                      |
+| v1.0    | careless user    | edit my expenses                                    | modify or correct the entries in my transaction history                               |
+| v1.0    | careless user    | delete my expenses                                  | remove incorrect or outdated entries from my transaction history                      |
+| v1.0    | responsible user | sort my expenses by date, name or amount            | have a clear timeline of how my budget has changed from the past to now               |
+| v1.0    | responsible user | add future payments                                 | budget accordingly based on expenses that will happen later on                        |
+| v1.0    | careless user    | edit future payments                                | modify or correct the entries of my upcoming expenses                                 |
+| v1.0    | careless user    | delete future payments                              | remove incorrect entries from my upcoming expenses                                    |
+| v1.0    | responsible user | view upcoming expenses within a period              | stay informed of the upcoming expenses within that time period and budget accordingly |
+| v1.0    | responsible user | pay future payments                                 | complete future payments and add the expenses to transaction history                  |
+| v1.0    | responsible user | remain informed of the remaining budget             | complete my future payments by managing other expenses                                |
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+
+  1. Program should work on any operating system supporting Java 11 or above. 
+  2. Does not require an active connection to the Internet to use the application.
+  3. Program should be accessible to users with no prior programming experience.
+  4. Program data will be saved persistently. 
+  5. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse. 
+
 
 ## Glossary
 
 * *glossary item* - Definition
 
-## Instructions for manual testing
-
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+## Instructions for Manual Testing
+### Launch
+1. Download the JAR file and move it into an empty folder.
+2. On a command line application, change the current working directory to the same folder as the JAR file and run the app using:\
+   `java -jar tp.jar`
+3. Expected: The app's welcome message is printed onto the terminal along with list of all commands.
```

Comment:
You can add some screenshot about how the apps start. It would make users to understand easily how they can start.